plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBSObjectContainer.java

```diff
@@ -93,4 +95,7 @@ public interface DBSObjectContainer extends DBSObject
      */
     void cacheStructure(@NotNull DBRProgressMonitor monitor, int scope) throws DBException;
 
+    default List<DBSObjectContainer> getImportedScopes(@NotNull DBRProgressMonitor monitor) throws DBException {

```

Comment:
Let's move it to separate interface DBS
DBSVisibiityScopeProvider {
  List<DBSStructContainer> getPublicContainers()
}
It should be implemented by datasource or catalog.

Almost all databases have INFORMATION_SCHEMA which is always visible. 
In PostgreSQL (and forks) we have searchPath (which we already read). It is dynamic and can be changed in runtime.